src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+ * <p>
+ * <h3>Implementation Requirements</h3>
+ * This class is immutable and thread-safe.
+ * <p>
+ * This class must be treated as a value type. Do not synchronize, rely on the
+ * identity hash code or use the distinction between equals() and ==.</p>
+ */
+public final class InternationalFixedDate
+        extends AbstractDate
+        implements ChronoLocalDate, Serializable {
+    /**
+     * Serialization version UID.
+     */
+    private static final long serialVersionUID = -908631218899744973L;
+    /**
+     * The days per 400 year cycle.
+     */
+    private static final int DAYS_PER_CYCLE = 146097;
+    /**
+     * Number of years in a decade.
+     */
+    private static final int YEARS_IN_DECADE = 10;
+    /**
+     * Number of years in a century.
+     */
+    private static final int YEARS_IN_CENTURY = 100;
+    /**
+     * Number of years in a millennium.
+     */
+    private static final int YEARS_IN_MILLENNIUM = 1000;
+    /**
+     * The proleptic year.
+     */
+    private final int prolepticYear;
+    /**
+     * The day of the year, it is calculated as (month - 1) * 28 + dayOfMonth + (isLeapYear() ? 1 : 0).
+     */
+    private final int dayOfYear;
+
+    /**
+     * Dates for both leap day and year day may only be instantiated through {@link InternationalFixedDate#of(int,boolean,boolean)}.
+     * It is therefore not possible to instantiated leap day as 2004-07-01 or year day as 2005-13-29.
+     *
+     * For calculation purposes, internally leap day is treated as day-of-year 169,
+     * squeezed between end of month 6 and beginning of month 7.
+     * Similarly, year day is treated as day-of-year 365 - or 366 in a leap year.
+     */
+
+    //-----------------------------------------------------------------------
+    /**
+     * Creates an instance from validated data.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from 1 to 13
+     * @param dayOfMonth    the International fixed day-of-month, from 1 to 28
+     * @return the International fixed date
+     */
+    private InternationalFixedDate(final int prolepticYear, final int month, final int dayOfMonth) {
+        this.prolepticYear = prolepticYear;
+        this.dayOfYear = (month - 1) * InternationalFixedChronology.DAYS_IN_MONTH + dayOfMonth + (isLeapYear() && month > 6 ? 1 : 0);
+    }
+
+    /**
+     * Creates an instance from validated data.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param isLeapDay     is it leap-day, month-less day between end of June and beginning of Sol
+     * @param isYearDay     is it year-day, month-less day between end of December of the old year and beginning of January of the following year
+     * @return the International fixed date
+     */
+    private InternationalFixedDate(final int prolepticYear, final boolean isLeapDay, final boolean isYearDay) {
```

Comment:
Boolean parameters are usually, but not always, an anti-pattern.  I'd prefer ditching this constructor for month/day-of-month or day-of-year.